src/components/data_submission/DataAccessGovernance.js

```diff
@@ -1,34 +1,88 @@
 import ConsentGroupForm from './consent_group/ConsentGroupForm';
 import { useEffect, useState, useCallback } from 'react';
-import { isNil, every, cloneDeep } from 'lodash/fp';
+import { isNil, every, cloneDeep, isEmpty, find } from 'lodash/fp';
 import { div, h, h2, a, span } from 'react-hyperscript-helpers';
-import { DAC } from '../../libs/ajax';
+import { DAC, DAR } from '../../libs/ajax';
 
 import './ds_common.css';
 
 export const DataAccessGovernance = (props) => {
   const {
-    onChange, onFileChange, validation, onValidationChange, setAllConsentGroupsSaved
+    onChange,
+    studyEditMode,
+    onFileChange,
+    validation,
+    datasets,
+    onValidationChange,
+    setAllConsentGroupsSaved
   } = props;
 
   const [consentGroupsState, setConsentGroupsState] = useState([]);
   const [dacs, setDacs] = useState([]);
 
+  const getDiseaseLabels = async (ontologyIds) => {
+    let labels = [];
+    if (!isEmpty(ontologyIds)) {
+      const idList = ontologyIds.join(',');
+      const result = await DAR.searchOntologyIdList(idList);
+      labels = result.map(r => r.label);
+    }
+    return labels;
+  };
+
+  const normalizeDataUse = useCallback(async (dataUse) => {
+    let du = dataUse;
+    if (!isNil(dataUse.diseaseRestrictions)) {
+      du.hasDiseaseRestrictions = true;
+      try {
+        du.diseaseLabels = await getDiseaseLabels(dataUse.diseaseRestrictions);
+      } catch {
+        // do nothing
+      }
+
+    }
+    if (!isNil(dataUse.other)) {
+      du.hasPrimaryOther = true;
+    }
+    if (!isNil(dataUse.secondaryOther)) {
+      du.hasSecondaryOther = true;
+    }
+    return du;
+  }, []);
+
+  const extractFileTypes = useCallback((propertyName, fileTypesName, dataset) => {
+    const property = find({ propertyName })(dataset.properties);
+    const fileTypesArr = [];
+
+    property?.propertyValue.forEach((propValue)=> {
+      fileTypesName === 'fileType' ?
+        fileTypesArr.push(propValue?.fileType)
+        : fileTypesArr.push(propValue?.functionalEquivalence);
+    });
+    return fileTypesArr;
+  }, []);
+
+  // method to extract consent group data from datasets
+  const extract = useCallback((propertyName, dataset) => {
+    const property = find({ propertyName })(dataset.properties);
+    return property?.propertyValue;
+  }, []);
+
   useEffect(() => {
     DAC.list(false).then((dacList) => {
       setDacs(dacList);
     });
   }, [setDacs]);
 
   useEffect(() => {
-    const filteredConsentGroupsState = consentGroupsState.filter(state => !isNil(state));
+    const filteredConsentGroupsState = consentGroupsState?.filter(state => !isNil(state));
```

Comment:
```suggestion
    const filteredConsentGroupsState = consentGroupsState.filter(state => !isNil(state));
```

Since on L176 you check if consentGroupState is empty, this should never be null or undefined.